format/Message.fbs

```diff
@@ -60,6 +92,9 @@ table RecordBatch {
   /// bitmap and 1 for the values. For struct arrays, there will only be a
   /// single buffer for the validity (nulls) bitmap
   buffers: [Buffer];
+
+  /// Optional compression of the message body
+  compression: RecordBatchCompression;
```

Comment:
What's the advantage of using the extra struct and 2 enums versus adding the properties `codec: CompressionType = NONE;` and `uncompressed_length: long = -1;` to the `struct Buffer` object directly? The latter doesn't need any special decoding path since the uncompressed length is in the metadata instead of an optional prefixed bytes, allows the producer to decide which buffers to compress: all, none, or a subset independently without an enum. And most importantly, is self-explanatory to anyone reading the struct as opposed to reading 2 structs and 2 enums and trying to guess their relations.

Reply:
Flatbuffer structs cannot be modified without breaking backwards compatibility

Reply:
Ah, this is very annoying.

Reply:
Yes. Also note that the size of these messages is meaningful. A `Buffer` struct currently takes up 16 bytes, so even if we _could_ add these fields, it wouldn't be a good idea because it would add metadata bloat even to messages that are not compressed. We aren't talking about large numbers relative to the size of the data, but in general I think we should try to keep the metadata as small as possible. In Protocol Buffers, optional fields don't take up space on the wire, so you can make different decisions there.